src/dos/drive_local.cpp

```diff
@@ -343,14 +343,29 @@ bool localDrive::FindNext(DOS_DTA & dta) {
 	//and due to its design dir_ent might be lost.)
 	//Copying dir_ent first
 	safe_strcpy(dir_entcopy, dir_ent);
-	if (stat(dirCache.GetExpandName(full_name),&stat_block)!=0) { 
+	char *temp_name = dirCache.GetExpandName(full_name);
+	if (stat(temp_name, &stat_block) != 0) {
 		goto again;//No symlinks and such
-	}	
+	}
+
+	if (stat_block.st_mode & S_IFDIR)
+		find_attr = DOS_ATTR_DIRECTORY;
+	else
+		find_attr = 0;
+#if defined(WIN32)
+	Bitu attribs = GetFileAttributes(temp_name);
+	if (attribs != INVALID_FILE_ATTRIBUTES)
+		find_attr |= attribs & 0x3f;
```

Comment:
Is it possible to name the 0x3f?
Suggest storing it in a `constexpr uint8_t ...` with a name for what it means.

Reply:
Done.